src/cpu/test_lockit/field.cc

```diff
@@ -323,6 +326,100 @@ void copyField(const TLColor src[][kHeight], TLColor dst[][kHeight])
     }
 }
 
+TLRensaResult simulate(TLColor field[][kHeight])
+{
+    // Assume all puyos are grounded.
+    // TODO: add DCHECK to check it.
+
+    int point[6][12] {};
+    int bottom[6] {};
+
+    // parameters necessary to compute score
+    int chain = 0;
+    int long_bonus[TLRensaResult::MAX_RENSA] {};
+    int num_colors[TLRensaResult::MAX_RENSA] {};
+    int num_connected[TLRensaResult::MAX_RENSA] {};   // # of connected puyos
+    int num_connections[TLRensaResult::MAX_RENSA] {};  // # of groups
+    bool quick = false;
+    
+    bool cont = true;
+    while (cont) {
+        cont = false;
+        memset(point, 0, sizeof(point));
```

Comment:
Instead of calling memset, define point here.
int point[6][12] {}; is usually faster than memset(point, 0, sizeof(point)); For small array, this is 2x faster.